slasher/service/data_update.go

```diff
@@ -37,3 +46,134 @@ func (s *Service) finalisedChangeUpdater() error {
 		}
 	}
 }
+
+func (s *Service) slasherOldAtetstationFeeder() error {
+	if s.beaconClient == nil {
+		return fmt.Errorf("can't feed old attestations to slasher. beacon client has not been started")
+	}
+	ch, err := s.beaconClient.GetChainHead(s.context, &ptypes.Empty{})
+	if err != nil {
+		log.Error(err)
+		return err
+	}
+	if ch.FinalizedEpoch < 2 {
+		log.Info("archive node doesnt have historic data for slasher to proccess. finalized epoch: %d", ch.FinalizedEpoch)
+	}
+	log.Infof("Finalized epoch is currently: %d", ch.FinalizedEpoch)
+	errOut := make(chan error)
+	var errorWg sync.WaitGroup
+	e, err := s.slasherDb.GetLatestEpochDetected()
+	if err != nil {
+		log.Error(err)
+		s.Stop()
+		return err
+	}
+	for i := e; i < ch.FinalizedEpoch; i++ {
+		ats, err := s.beaconClient.ListAttestations(s.context, &ethpb.ListAttestationsRequest{
+			QueryFilter: &ethpb.ListAttestationsRequest_TargetEpoch{TargetEpoch: i},
+		})
+		if err != nil {
+			log.Error(err)
+		}
+		bcs, err := s.beaconClient.ListBeaconCommittees(s.context, &ethpb.ListCommitteesRequest{
+			QueryFilter: &ethpb.ListCommitteesRequest_Epoch{
+				Epoch: i,
+			},
+		})
+		if err != nil || bcs == nil {
+			log.Error(err)
+			continue
+		}
+		log.Infof("detecting slashable events on: %v attestations from epoch: %v", len(ats.Attestations), i)
+		for _, attestation := range ats.Attestations {
+			scs, ok := bcs.Committees[attestation.Data.Slot]
+			if !ok {
+				var keys []uint64
+				for k := range bcs.Committees {
+					keys = append(keys, k)
+				}
+				log.Errorf("committees doesnt contain the attestation slot: %d, actual first slot: %v", attestation.Data.Slot, keys)
+				continue
+			}
+			if attestation.Data.CommitteeIndex > uint64(len(scs.Committees)) {
+				log.Errorf("committee index is out of range in slot wanted: %v, actual", attestation.Data.CommitteeIndex, len(scs.Committees))
+				continue
+			}
+			sc := scs.Committees[attestation.Data.CommitteeIndex]
+			c := sc.ValidatorIndices
+			ia, err := ConvertToIndexed(s.context, attestation, c)
+			if err != nil {
+				log.Error(err)
+				continue
+			}
+			sar, err := s.slasher.IsSlashableAttestation(s.context, ia)
+			if err != nil {
+				log.Error(err)
+				continue
+			}
+			s.slasherDb.SaveAttesterSlashings(db.Active, sar.AttesterSlashing)
+			if len(sar.AttesterSlashing) > 0 {
+				log.Infof("slashing response: %v", sar.AttesterSlashing)
```

Comment:
Use better-structured and formatted logging here as well

Reply:
done